.github/workflows/docker-build-and-test.yml

```diff
@@ -58,33 +60,212 @@ jobs:
         id: get-matrix
         run: |
           # shellcheck disable=SC2046
-          echo matrix=$( yq '.services | keys[] | select(. | test("coreboot.*"))' docker/compose.yaml | sed 's/"//g' | jq -Rs 'split("\n") | del(.[-1])' | jq -c ) >> "${GITHUB_OUTPUT}"
+          echo matrix=$( yq --input-format yaml --output-format json '.services | keys[] | select(. | test("coreboot.*"))' docker/compose.yaml | sed 's/"//g' | jq -Rs 'split("\n") | del(.[-1])' | jq -c ) >> "${GITHUB_OUTPUT}"
+      - name: Check
+        run: |
+          jq . <<< '${{ steps.get-matrix.outputs.matrix }}'
     outputs:
       matrix: ${{ steps.get-matrix.outputs.matrix }}
 
-  get-matrix-edk2:
-    runs-on: ubuntu-latest
+  #=====================
+  # Coreboot toolchains
+  #=====================
+
+  build-coreboot-toolchains:
+    timeout-minutes: 120
+    needs:
+      - get-matrix-coreboot
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: ['amd64', 'arm64']
+        dockerfile: ${{ fromJson(needs.get-matrix-coreboot.outputs.matrix) }}
+    runs-on: ${{ matrix.arch == 'arm64' && 'ARM64' || 'ubuntu-latest' }}
+    container:
+      # At the time of writing (2024-10) we cannot use ubuntu:noble as it is broken
+      image: ubuntu:jammy
+    env:
+      DEBIAN_FRONTEND: noninteractive
+      # Use coreboot mirrors
+      BUILDGCC_OPTIONS: -m
     steps:
+      - name: Install dependencies for CI
+        run: |
+          apt-get update
+          apt-get install -y --no-install-recommends \
+            ca-certificates \
+            curl \
+            git \
+            jq \
+            sudo \
+            tzdata \
+            upx-ucl \
+            wget
+          update-ca-certificates
+      - name: Get yq
```

Comment:
I'm a bit confused here. So you fetch `yq` in this step which makes it available for this jobs following steps but inside the get-matrix(-coreboot) jobs you also call into `yq` without fetching it first, implying that it was already present. Was it not present on jammy or am I missing something here?

Reply:
huh, that is a good question ...

Reply:
According to [github runner-images](https://github.com/actions/runner-images?tab=readme-ov-file#available-images) `ubuntu-latest` == `22.04` == `jammy`.

So technically speaking everything runs on `jammy` ...

Reply:
So I guess fetching yq is not needed then or do the two matrix jobs not work right now?

Reply:
Oh yeah, there is a difference. `runs-on:` is usign GitHub's flavoured Docker containers where `yq` [is installed](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#tools), but `container:` is [using upstream Docker-hub images](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses) which are minimal.

Reply:
`get-matrix` and `get-matrix-coreboot` are using `runs-on: ubuntu-latest` which has `yq` installed, but `build-coreboot-toolchains` runs on various GitHubs ubuntu containers (based on config in `docker/compose.yaml` it is only `jammy` at the moment) but then they spawn a Docker container `container: image: ubuntu:jammy` which is from Docker-hub without `yq`.

Reply:
I see, thanks for clarifying!